lib/streamlit/type_util.py

```diff
@@ -147,6 +147,10 @@ def is_array_value_field_name(obj: object) -> TypeGuard[ArrayValueFieldName]:
     return obj in ARRAY_VALUE_FIELD_NAMES
 
 
+def check_type(val: Any) -> type:
```

Comment:
I think this function makes the internal typing a bit worse. Just using `type(val) is str` is giving mypy the information that `val` has to be string. I think with `check_type`, this is not working (however, I haven't tested it).

I would just directly use `isinstance(s, str)` above to check the type.

Reply:
@LukasMasuch true, should have use `isinstance` right away! Removed the func and replaced `check_type` with `isintance` check, thank you 🙂